### PR TITLE
Make all FastItemAdapter functions opened

### LINKED ISSUE #776: Mark functions as opened in FastItemAdapter
v4.0.0

Currently, `FastItemAdapter` is opened, meaning that it can be extended.
However, all of its functions are final, so it isn't extremely useful.
Consider marking functions as `open` as well so that they can be overridden.

As an example, I used to have a `FastItemThemedAdapter`, there it would modify the items before they were added.

### PR DESCRIPTION
**Pull request by AllanWang**
Resolves #776 

You can also open `itemAdapter`, though it leads to the warning of accessing a non final property during `init`. There is already a problem with this through `addAdapter`, which you may want to look into, but that is a separate issue

### CODE REVIEW

**Comment by mikepenz** [comment]
@AllanWang yes I know about that warning for non final properties. :/

I think the `ItemAdapter` and `ModelAdapter` should be open too yes.

**Comment by AllanWang** [comment]
One fix is to put those into an actual function `init()` and then call

```kt
init {
  init()
}
```

It does get rid of the warnings, but I wonder if it actually makes a difference

**Comment by AllanWang** [comment]
Opened model adapter. I remove `operator` from `set` as I again don't think it's the right style.

File: library-core/src/main/java/com/mikepenz/fastadapter/adapters/ModelAdapter.kt [hunk 1]
```diff
@@ -187,7 +187,7 @@ open class ModelAdapter<Model, Item : IItem<out RecyclerView.ViewHolder>>(
      * @param adapterNotifier a `IAdapterNotifier` allowing to modify the notify logic for the adapter (keep null for default)
      * @return this
      */
-    operator fun set(
```
Comment (FabianTerhorst) [inline library-core/src/main/java/com/mikepenz/fastadapter/adapters/ModelAdapter.kt]: Breaking API change. Not sure if this is better without.

**Pull request merged**